datasets/cord19/cord19.py

```diff
@@ -0,0 +1,207 @@
+        datasets.BuilderConfig(
+            name="metadata",
+            description="The set of documents but loading some metadata like title and " "abstract for each article.",
+        ),
+        datasets.BuilderConfig(
+            name="fulltext",
+            description="The set of documents loading some metadata like title and "
+            "abstract and full text for each article.",
+        ),
+        datasets.BuilderConfig(
+            name="embeddings",
+            description="The set of documents loading some metadata like title and "
+            "abstract and document embeddings for each article.",
+        ),
+    ]
+
+    def _info(self):
+        # default metadata only
+        features_dict = {
+            "cord_uid": datasets.Value("string"),
+            "sha": datasets.Value("string"),
+            "source_x": datasets.Value("string"),
+            "title": datasets.Value("string"),
+            "doi": datasets.Value("string"),
+            "abstract": datasets.Value("string"),
+            "publish_time": datasets.Value("string"),
+            "authors": datasets.Value("string"),
+            "journal": datasets.Value("string"),
+            "url": datasets.Value("string"),
+        }
+
+        if "fulltext" in self.config.name:
+            # adding full_text
+            features_dict["fulltext"] = datasets.Value("string")
+
+        if "embeddings" in self.config.name:
+            # adding embeddings
+            features_dict["doc_embeddings"] = datasets.Sequence(datasets.Value("float64"))
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=datasets.Features(features_dict),
+            supervised_keys=None,
+            homepage="https://www.semanticscholar.org/cord19/download",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        my_urls = _URL
+        data_dir = dl_manager.download_and_extract(my_urls)
+
+        files = dict()
+        files["metadata"] = os.path.join(data_dir, "2020-11-29/metadata.csv")
+
+        if "fulltext" in self.config.name:
+            fulltext_dir_path = os.path.join(data_dir, "2020-11-29/document_parses")
+            if not os.path.isdir(fulltext_dir_path):
+                shutil.unpack_archive(os.path.join(data_dir, "2020-11-29/document_parses.tar.gz"), fulltext_dir_path)
+            files["fulltext"] = fulltext_dir_path
+
+        if "embeddings" in self.config.name:
+            embeddings_dir_path = os.path.join(data_dir, "2020-11-29/cord_19_embeddings")
+            if not os.path.isdir(embeddings_dir_path):
+                shutil.unpack_archive(
+                    os.path.join(data_dir, "2020-11-29/cord_19_embeddings.tar.gz"), embeddings_dir_path
+                )
```

Comment:
Can you use `dl_manager.extract` instead ?

Reply:
Tested locally: there is only 2 files to untar and it seems much slower than with shutil (roughly 5 times slower).

Reply:
Interesting ! Maybe we can use shutil to improve the speed of our file extraction.

Can you still change for dl_manager.extract ? The dl_manager is responsible for handling cache files.